runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -0,0 +1,674 @@
+
+    /** {@inheritDoc} */
+    @Override
+    @TransactionalMethod
+    public V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V>
+            remappingFunction) {
+        Objects.requireNonNull(remappingFunction);
+        V oldValue;
+        if ((oldValue = get(key)) != null) {
+            V newValue = remappingFunction.apply(key, oldValue);
+            if (newValue != null) {
+                put(key, newValue);
+                return newValue;
+            } else {
+                remove(key);
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @TransactionalMethod
+    public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V>
+            remappingFunction) {
+        Objects.requireNonNull(remappingFunction);
+        Objects.requireNonNull(value);
+        V oldValue = get(key);
+        V newValue = (oldValue == null) ? value :
+                remappingFunction.apply(oldValue, value);
+        if (newValue == null) {
+            remove(key);
+        } else {
+            put(key, newValue);
+        }
+        return newValue;
+    }
+
+    /** Unmaps the secondary indexes for a given key value pair.
+     *
+     * @param key           The primary key (index) for the mapping.
+     * @param value         The value to unmap.
+     */
+    @SuppressWarnings("unchecked")
+    protected void unmapSecondaryIndexes(K key, V value) {
+        try {
+            if (value != null) {
+                final Map.Entry<K, V> previousEntry
+                        = new AbstractMap.SimpleImmutableEntry<>(key, value);
+                indexFunctions.parallelStream()
+                        .forEach(f -> f.getIndexFunction().generateIndex(key, value)
+                                .forEach(i -> indexMap.get(f).remove((I) i, previousEntry)));
+
+            }
+        } catch (Exception e) {
+            indexFunctions = Collections.emptySet();
+            indexMap.clear();
+            indexGenerationFailed = true;
+            log.error("unmapSecondaryIndexes: Exception unmapping {}, {},"
+                            + " UNMAPPING ALL INDEXES, indexing is disabled",
+                    key, value, e);
+        }
+    }
+
+    /** Maps the secondary indexes for a given key value pair.
+     *
+     * @param key       The primary key (index) for the mapping.
+     * @param value     The value to map.
+     */
+    @SuppressWarnings("unchecked")
+    protected void mapSecondaryIndexes(K key, V value) {
+        try {
+            Map.Entry<K, V> entry = new AbstractMap.SimpleImmutableEntry<>(key, value);
+            indexFunctions.parallelStream()
+                    .forEach(f -> f.getIndexFunction().generateIndex(key, value)
+                            .forEach(i -> indexMap.get(f).put((I) i, entry)));
+        } catch (Exception e) {
+            indexFunctions = Collections.emptySet();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)